src/main/java/gift/domain/product/controller/ProductController.java

```diff
@@ -0,0 +1,102 @@
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductDao productDao;
+
+    public ProductController(ProductDao productDao) {
+        this.productDao = productDao;
+    }
+
+    @GetMapping("/new")
+    public String renderingNewForm() {
+        return "new-product";
+    }
+
+    @GetMapping("/edit/{productId}")
+    public String renderingEditForm(@PathVariable long productId, Model model) {
+        Optional<Product> product = productDao.findById(productId);
+
+        if (product.isEmpty()) {
+            return "error";
+        }
+        model.addAttribute("product", product.get());
+
+        return "edit-product";
+    }
+
+    @PostMapping
+    public String create(@ModelAttribute ProductDto productDto) {
+        Product product = productDto.toProduct();
+
+        Product savedProduct = productDao.insert(product);
+
+        return "redirect:/products/" + savedProduct.getId();
+    }
+
+    @GetMapping
+    public String readAll(Model model) {
+        List<Product> productList = productDao.findAll();
+        model.addAttribute("products", productList);
+        return "products";
+    }
+
+    @GetMapping("/{productId}")
+    public String readById(@PathVariable long productId, Model model) {
+        Optional<Product> product = productDao.findById(productId);
+
+        if (product.isEmpty()) {
+            return "error";
+        }
+        model.addAttribute("product", product.get());
+
+        return "product";
+    }
+
+    @PutMapping("/{productId}")
+    public String update(@PathVariable long productId, @ModelAttribute ProductDto productDto) {
+        Product product = productDto.toProduct();
+        product.setId(productId);
+
+        Optional<Product> updatedProduct = productDao.update(product);
+
+        if (updatedProduct.isEmpty()) {
+            return "error";
+        }
+
+        return "redirect:/products/" + updatedProduct.get().getId();
+    }
+
+    @DeleteMapping("/{productId}")
+    public ResponseEntity<Void> delete(@PathVariable long productId) {
+        int nOfRowsAffected = productDao.delete(productId);
+
+        if (nOfRowsAffected != 1) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
```suggestion
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }

        return ResponseEntity.noContent()
```
와 같이 빌더 패턴을 쓰는걸 추천드려요!